### PR TITLE
[BUG] : fixes org.opensearch.search.aggregations.metrics.MedianAbsoluteDeviationAggregatorTests.testQueryFiltering test failure 

### PR DESCRIPTION
**Pull request by SohamRatnaparkhi**
### Description
[Describe what this change achieves]
Fixes the test failure, possibly flakey - Noticed in #4789 
### Issues Resolved
[List any issues this PR will resolve]
Solves issue #4789 
All commits have been Signed-off-by Soham Ratnaparkhi
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [X] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by tlfeng** [comment]
About adding sign-off message, you could either add `-s` flag in `git commit` command, or manually append the following message to your commit message "Signed-off-by: your_name <email_address>".

When forgot adding sign-off message in a middle commit, there are 2 ways:
1 Squash all the above commits and then append sign-off message to the squashed commit.
2 Modify each commit in the PR that doesn't have sign-off message.

**Comment by adnapibar** [comment]
@SohamRatnaparkhi can you rebase and update your PR?

**Comment by Poojita-Raj** [comment]
Hey @SohamRatnaparkhi, could you update this PR and rebase the changes to deal with the branch conflict?

File: server/src/test/java/org/opensearch/search/aggregations/metrics/MedianAbsoluteDeviationAggregatorTests.java [hunk 1]
```diff
@@ -238,7 +238,7 @@ private void testAggregation(
         Consumer<InternalMedianAbsoluteDeviation> verify
     ) throws IOException {
         MedianAbsoluteDeviationAggregationBuilder builder = new MedianAbsoluteDeviationAggregationBuilder("mad").field(FIELD_NAME)
-            .compression(randomDoubleBetween(20, 1000, true));
+            .compression(randomDoubleBetween(21, 1000, true));
```
Comment (Poojita-Raj) [inline server/src/test/java/org/opensearch/search/aggregations/metrics/MedianAbsoluteDeviationAggregatorTests.java:241]: Hi, could you mention in a comment why this value change would fix this issue?

**Comment by Poojita-Raj** [comment]
Gradle check failure: org.opensearch.indices.replication.SegmentReplicationAllocationIT.testAllocationWithDisruption

**Comment by andrross** [comment]
Superseded by #7317

**Pull request closed without merging**